src/GNDStk/Node/src/add-child.hpp

```diff
@@ -0,0 +1,283 @@
+
```

Comment:
Another issue with this is there are many different functions with every-so-slightly-different signatures. Do we really have to have different definitions of this? Again, this could be an issue with my understanding of the problem, but it feels unnecessarily complex.

Reply:
I can't really argue too much with this. One reason I split Node's `add()` functions into two files (for metadata adds, and for child-node adds) was to try to help, if only a little, with all of it making sense. The plethora of `add()` variations makes the capability very flexible and powerful; unfortunately, that came at the price of a non-trivial amount of SFINAE drain bramage in order to make various calls overload to the right place. (Not the wrong place, which would trigger other problems and result in the spectacular - in a bad way - compiler error messages that we all know and love.) I'm hoping that the flexibility and power of the `add()` functions will be worth it. And, again, the upcoming documentation should be and to help in this respect.